lib/qm-dsp/dsp/tempotracking/DownBeat.h

```diff
@@ -53,35 +53,15 @@ class DownBeat
 
     void setBeatsPerBar(int bpb);
 
-    /**
-     * Estimate which beats are down-beats.
-     * 
-     * audio contains the input audio stream after downsampling, and
-     * audioLength contains the number of samples in this downsampled
-     * stream.
-     *
-     * beats contains a series of beat positions expressed in
-     * multiples of the df increment at the audio's original sample
-     * rate, as described to the constructor.
-     *
-     * The returned downbeat array contains a series of indices to the
-     * beats array.
-     */
-    void findDownBeats(const float *audio, // downsampled
+    
+    // Return the beat spectral difference function. The returned
+    // vector contains one value for each of the beats, less one.
+    // Each value contains the spectral difference between region prior
+    // to the beat's nominal position and the region following it.    
```

Comment:
Why is this done with a new FFT? I think we already have the FFT from the beat detection ...

Reply:
Here the audio has been downsampled by a factor of 16. Targeted at 3khz.
Also, the FFT here has a step of beat length rather than a fixed one.
I think we can actually consider using this, for the beat detection too, right?